docs/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to.md

```diff
@@ -171,6 +220,150 @@ public class ExampleClass
+
+For that functionality, you can write a custom converter. Here's a sample POCO and a custom converter for it that illustrates this approach:
+
+[!code-csharp[](snippets/system-text-json-how-to/csharp/WeatherForecast.cs?name=SnippetWF)]
+
+[!code-csharp[](snippets/system-text-json-how-to/csharp/WeatherForecastRuntimeIgnoreConverter.cs)]
+
+The converter causes the `Summary` property to be omitted from serialization if its value is null, an empty string, or "N/A".
+
+Register this custom converter by [using an attribute on the class](system-text-json-converters-how-to.md#registration-sample---jsonconverter-on-a-type) or by [adding the converter](system-text-json-converters-how-to.md#registration-sample---converters-collection) to the <xref:System.Text.Json.JsonSerializerOptions.Converters> collection.
+
+This approach requires additional logic if:
+
+* The POCO includes complex properties.
+* You need to handle attributes such as `[JsonIgnore]` or options such as custom encoders.
+
+### Public and non-public fields
+
+`Newtonsoft.Json` can serialize and deserialize fields as well as properties.
+
+::: zone pivot="dotnet-5-0"
+In <xref:System.Text.Json>, use the <xref:System.Text.Json.JsonSerializerOptions.IncludeFields?displayProperty=nameWithType> global setting to include public fields when serializing or deserializing. For an example, see [Include fields](system-text-json-how-to.md#include-fields).
+::: zone-end
+
+::: zone pivot="dotnet-core-3-1"
+<xref:System.Text.Json> 3.1 only works with public properties. Custom converters can provide this functionality.
+::: zone-end
+
+### Preserve object references and handle loops
+
+By default, `Newtonsoft.Json` serializes by value. For example, if an object contains two properties that contain a reference to the same `Person` object, the values of that `Person` object's properties are duplicated in the JSON.
+
+`Newtonsoft.Json` has a `PreserveReferencesHandling` setting on `JsonSerializerSettings` that lets you serialize by reference:
+
+* An identifier metadata is added to the JSON created for the first `Person` object.
+* The JSON that is created for the second `Person` object contains a reference to that identifier instead of property values.
+
+`Newtonsoft.Json` also has a `ReferenceLoopHandling` setting that lets you ignore circular references rather than throw an exception.
+
+::: zone pivot="dotnet-5-0"
+To preserve references and handle circular references in <xref:System.Text.Json>, set <xref:System.Text.Json.JsonSerializerOptions.ReferenceHandler%2A?displayProperty=nameWithType> to <xref:System.Text.Json.Serialization.ReferenceHandler.Preserve%2A>. The `ReferenceHandler.Preserve` setting is equivalent to `ReferenceLoopHandling` = `true` and `PreserveReferencesHandling` = `true` in `Newtonsoft.Json`.
+
+Some related `Newtonsoft.Json` features are not supported:
+
+* [ReferenceResolver](https://www.newtonsoft.com/json/help/html/P_Newtonsoft_Json_JsonSerializer_ReferenceResolver.htm)
```

Comment:
We expose our own `ReferenceResolver` type which allows folks to implement custom preserve handling logic.